manifests/database/postgresql.pp

```diff
@@ -10,20 +13,111 @@
     default => postgresql_password($::foreman::db_username, $::foreman::db_password),
   }
 
+  $rep_password = $::foreman::db_replication_password ? {
+    'UNSET' => false,
+    default => postgresql_password($::foreman::db_replication_username, $::foreman::db_replication_password),
+  }
+
+  $port = $::foreman::db_port ? {
+    'UNSET' => 5432,
+    default => $::foreman::db_port,
+  }
+
   # Prevents errors if run from /root etc.
   Postgresql_psql {
     cwd => '/',
   }
 
-  include ::postgresql::client, ::postgresql::server
+  if $::foreman::db_cluster {
+    $db_all_cluster_hostnames = concat($::foreman::db_cluster_hostnames, $::fqdn)
+    if $::foreman::db_node_type == 'master' {
+      class {'postgresql::server':
```

Comment:
We try very hard not to manage external classes so users can override them in profiles.